### DISCUSSION TITLE
how to open a view when i click on a action

### DISCUSSION THREAD

**Discussion by abhi051002** [filamentphp/filament#10014] [Help]
i have action type Filament\Tables\Actions\Modal\Actions\Action; inside of a  Filament\Tables\Actions\ViewAction i want to return/redirect a view file but iwas unable to do it. when i use modalContent it gives not available error. if anyone know how to do it please help me

**Comment by ArtMin96** [comment]
[Action modals](https://filamentphp.com/docs/3.x/actions/modals)
```php
Action::make('action')
    ->view('path.to.view')
```
or
```php
->modalContent(view('path.to.view'))
```
?

**Reply by abhi051002** [reply]
Thank you so much but i want to open a modal when i click that button
->modalContent(view('path.to.view')) it shows Method Filament\Tables\Actions\Modal\Actions\Action::modalContent does not exist.
